modules/persistence/index.ts

```diff
@@ -38,6 +38,9 @@ const app = async (path: string, githubUser: string) => {
     // that only one build will be used per run of this module.
     const buildId = new mongodb.ObjectId();
     const metadata = await metadataFromZip(zip, githubUser);
+    // initialize db connections to handle shared connections
+    await snootyDb();
+    await poolDb();
```

Comment:
Since the 2 dbs are initiated from the same client connection, I think it might be safe to just initiate one of them. But no worries if we want to keep both for safety

Reply:
Oh, but does that still work if they are connecting to different databases in these different functions?

Reply:
From what I could tell, yes? I commented out one of them, rebuilt, then reran the module, and it seemed like the result was the same. But again, I'm totally okay with keeping both for safety

Reply:
When I run it locally, one of them (poolDb) is never called. So, no matter what you won't see multiple connections locally from the poolDb.

Reply:
I really should learn more about this database called MongoDB Atlas :)

Reply:
Hm, I'd imagine `poolDb` would still eventually be called by one of the functions further down the module. But all good. Down to keep as-is to avoid any weird behavior with connections 😨

Reply:
these still might have a [race condition (?) ](https://github.com/mongodb/node-mongodb-native/blob/v4.9.1/src/operations/connect.ts#L18-L19) if the topology is making simultaneous connections?

think ideally we would call client.connect once

Reply:
Both dbs are referenced through the same/single client, so as long as `client.connect` is called once (by `await snootyDb()`, for example) before everything else, I don't think there would be a race condition in simultaneous connections. Any future calls to respective dbs should just attempt to connect, see it's already connected, and then set the `dbInstance` as needed